src/webview/message/handler.ts

```diff
@@ -173,27 +182,50 @@ export class MessageToWebviewHandlerImpl implements MessageToWebviewHandler {
     return extractNumberFromString(lineNumberElement.textContent);
   }
 
-  private hideViewedFiles(diffContainer: HTMLElement, viewedState: ViewedState) {
+  private async hideViewedFiles(diffContainer: HTMLElement, viewedState: ViewedState) {
     const viewedToggles = diffContainer.querySelectorAll<HTMLInputElement>(
       Diff2HtmlCssClassElements.Input__ViewedToggle
     );
     for (const toggle of Array.from(viewedToggles)) {
       const fileName = this.getDiffElementFileName(toggle);
-      if (fileName && viewedState[fileName]) toggle.click();
+      if (fileName && viewedState[fileName]) {
+        const diffHash = await this.getDiffHash(toggle);
+        if (diffHash === viewedState[fileName]) {
+          toggle.click();
+        } else {
+          toggle.classList.add(CHANGED_SINCE_VIEWED);
+        }
+      }
     }
   }
 
-  private sendFileViewedMessage(toggleElement: HTMLInputElement): void {
+  private async getDiffHash(diffElement: HTMLElement) {
```

Comment:
it is the toggle element in both places where `getDiffHash` gets called, but the function doesn't care – it can be given any element inside a diff - this is similar to other users of `diffElement` in the file.